examples/kubeflow/mnist/pipeline.py

```diff
@@ -82,21 +88,124 @@ def kubeflow_pipeline(s3_endpoint: str, input_bucket: str):
-        }
-    ).run_id
+parser = argparse.ArgumentParser(description='Create a kubeflow pipeline ' +
+    'that trains a small neural net on MNIST data')
+parser.add_argument('--remote_host',
+                    default="", metavar="http://host/path", type=str,
+                    help='The address of the kubeflow pipelines API, if '+
+                         'running outside a kubeflow cluster')
+parser.add_argument('--create_run_in',
+                    default="", metavar="pipeline_name", type=str,
+                    help='If set, this script will create a run for the '+
+                    'given kubeflow pipeline and wait for it to complete, '+
+                    'rather than uploading a new pipeline')
+parser.add_argument('--create_pipeline',
+                    default="", metavar="pipeline_name", type=str,
+                    help='If set, this script will create a kubeflow pipeline '+
+                    'with the given name, but will not create a run in that '+
+                    'pipeline. Used to create a pipeline from a local machine.')
+parser.add_argument('--force', action='store_true', default=False,
+                    help='If set, and this script tries to create a pipeline '+
+                    'where a kubeflow pipeline with the same name already '+
+                    'exists, the existing pipeline will be deleted')
+
+def pipeline_id(client : kfp.Client, name : str):
+    """Gets the ID of the kubeflow pipeline with the name 'name'
+    Args:
+      name of the pipeline
+    Returns:
+      id of the pipeline
+    """
+    page_token = ""
+    while True:
+        p = client.list_pipelines(page_token=page_token, page_size=100)
+        if p.pipelines is None:
+            return ""
+        for p in p.pipelines:
+            if p.name == name:
+                return p.id
+        if p.next_page_token is None:
+            return ""
+        page_token = p.next_page_token
+
+def experiment_id(client : kfp.Client, name : str):
+    """Gets the ID of the kubeflow experiment with the name 'name'
+    Args:
+      name of the experiment
+    Returns:
+      id of the experiment
+    """
+    page_token = ""
+    while True:
+        p = client.list_experiments(page_token=page_token, page_size=100)
+        for p in p.experiments:
+            if p.name == name:
+                return p.id
+        if p.next_page_token is None:
+            return ""
+        page_token = p.next_page_token
 
-    print("run id: {}".format(run_id))
-
-    j = client.wait_for_run_completion(run_id, 60)
-    assert j.run.status == 'Succeeded'
+def main():
+    args = parser.parse_args()
+    if args.remote_host != "":
+        client = kfp.Client(host=args.remote_host)
+    else:
+        client = kfp.Client()
+
+    run_id = ""
+    if args.create_run_in != "" and args.create_pipeline != "":
+        logging.error('only one of --create_run_in and '+
+              '--create_pipeline may be set')
+        sys.exit(1)
```

Comment:
Rather than calling sys.exit, have main return the process' rc, then below do this:

```
if __name__ == "__main__":
    sys.exit(main())
```

Reply:
Done